internal/priorityqueue/priorityqueue.go

```diff
@@ -19,8 +19,8 @@ type PriorityQueue[T any] struct {
 // compares the priorities of two items a and b; it should return a number > 0
 // if the priority of a is higher, 0 if the priorities are equal a number < 0
 // otherwise.
-func New[T any](cmp func(a, b T) int) *PriorityQueue[T] {
-	return &PriorityQueue[T]{cmp: cmp, items: make([]T, 1)}
+func New[T any](sz int, cmp func(a, b T) int) *PriorityQueue[T] {
```

Comment:
Call it `sizeHint`, if -1 is passed it should be ignored, and the comment on `New` should explain this

Reply:
OK

Reply:
Done